From d0982d77e87baafbe25d2101426dedf9498e7759 Mon Sep 17 00:00:00 2001
From: Laurent Pinchart <laurent.pinchart@ideasonboard.com>
Date: Thu, 23 May 2013 20:40:29 +0200
Subject: [PATCH 311/509] dma: xilinx: vdma: Simplify interrupt handler

https://github.com/analogdevicesinc/linux.git xcomm_zynq
commit 6d6c2dce41583bf87fae3f888282793ce7e95613

Perform actions directly instead of setting intermediate variables only
to test them right afterwards.

Signed-off-by: Laurent Pinchart <laurent.pinchart@ideasonboard.com>
Acked-by: Srikanth Thokala <sthokal@xilinx.com>
Signed-off-by: Zhong Hongbo <hongbo.zhong@windriver.com>
---
 drivers/dma/xilinx/xilinx_axivdma.c |   10 +---------
 1 files changed, 1 insertions(+), 9 deletions(-)

diff --git a/drivers/dma/xilinx/xilinx_axivdma.c b/drivers/dma/xilinx/xilinx_axivdma.c
index 5fbcceee..8a2b233 100644
--- a/drivers/dma/xilinx/xilinx_axivdma.c
+++ b/drivers/dma/xilinx/xilinx_axivdma.c
@@ -604,8 +604,6 @@ static int xilinx_vdma_reset(struct xilinx_vdma_chan *chan)
 static irqreturn_t xilinx_vdma_irq_handler(int irq, void *data)
 {
 	struct xilinx_vdma_chan *chan = data;
-	int update_cookie = 0;
-	int to_transfer = 0;
 	u32 stat, reg;
 
 	reg = vdma_ctrl_read(chan, XILINX_VDMA_REG_DMACR);
@@ -656,15 +654,9 @@ static irqreturn_t xilinx_vdma_irq_handler(int irq, void *data)
 		dev_dbg(chan->dev, "Inter-packet latency too long\n");
 
 	if (stat & XILINX_VDMA_DMASR_FRM_CNT_IRQ) {
-		update_cookie = 1;
-		to_transfer = 1;
-	}
-
-	if (update_cookie)
 		xilinx_vdma_update_completed_cookie(chan);
-
-	if (to_transfer)
 		xilinx_vdma_start_transfer(chan);
+	}
 
 	tasklet_schedule(&chan->tasklet);
 	return IRQ_HANDLED;
-- 
1.7.5.4

